bokehjs/src/coffee/models/plots/plot_canvas.coffee

```diff
@@ -402,7 +402,12 @@ class PlotCanvasView extends Renderer.View
     # reach the frame in time (sometimes) so force an update here for now
     @model.get('frame')._update_mappers()
 
+    # Get context and restore to its initial state. The restoring
+    # Would stricly speaking not be necessary, but in case there is an error
+    # during drawing, restore() won't be called enough. This guards for that.
     ctx = @canvas_view.ctx
+    for i in [0..3]
+      ctx.restore()
```

Comment:
I understand the idea,  my question is this: how are we assured that previous intentional ctx `save` operations that might have happened prior to / outside this area of code aren't blown away. Let's say that code that calls this render does a `ctx.save`, and then calls this render, and there are no errors. Then this code calls `ctx.restore` three times, unconditionally? I don't see how that earlier `ctx.save` does not get blown away, even though it should not be.